src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -22,7 +22,6 @@ def ingest(
     source: Union[Sequence[str], str],
     output: Union[Sequence[str], str],
     config: Mapping[str, Any],
-    access_credentials_name: str,
```

Comment:
This should still be defined as a parameter to the function. Let's not hide things behind generic kwargs.

Reply:
I believe all that is required is setting the type to.optional with a default of None?

Reply:
Nope this is not sufficient. Type checking in python is nothing more than just syntax sugar. The alternative I pushed is to keep it as a non positional argument and pop from the wrapper instead of just getting it.

Reply:
Type checking is used by language servers and linters, I'd like us to keep the argument as a parameter. It can be handled, we do this in other functions. Please consult with @thetorpedodog on how we can handle this.

Reply:
Defining it this way means it remains a full-fledged parameter, just one that _has_ to be passed as a named argument. Language servers and linters are able to handle this, and moving the name after `*args` is the right way to accomplish it.

```python

def some_func(first, this_arg, *args, that_arg="z"): ...

# In both of these cases, first="f", this_arg="a", args=(1, 2, 3), that_arg="z".
some_func("f", "a", 1, 2, 3)
some_func("f", 1, 2, 3, this_arg="a")

# first="f", this_arg=1, args=(2, 3), that_arg="b"
some_func("f", 1, 2, 3, that_arg="b")

def other_func(first, *args, this_arg, that_arg="z"): ...

# Neither of these is allowed, since `this_arg` must be explicitly named.
other_func("f", "a", 1, 2, 3)
other_func("f", 1, 2, 3, that_arg="b")

# This one works, with first="f", this_arg="a", args=(1, 2, 3), that_arg="z"
other_func("f", 1, 2, 3, this_arg="a")
```

Reply:
In my last commit I returned the access_credentials_name as a named argument as @Shelnutt2 recommended. My answer to @Shelnutt2 was about the problem that setting it to `Optional` or setting a default value does not resolve the issue I am trying to solve. Hence I added the `pop`. 
```python
import inspect
def some_func(first, this_arg, *args, that_arg="z"):
    print(a)
valid_args = inspect.signature(some_func).parameters
valid_args
>>>mappingproxy({'first': <Parameter "first">,
              'this_arg': <Parameter "this_arg">,
              'args': <Parameter "*args">,
              'that_arg': <Parameter "that_arg='z'">})
```
As you notice even when passed as a named argument it is returned from the `inspect` and this where the issue lies. _filter_dict function does not filter out the `access_credentials_name` and this is fed to `graph.submit`, where we already explicitly set the `access_credentials_name`.